src/modules/cpus/rv32-imac/riscv32.zig

```diff
@@ -0,0 +1,921 @@
+pub const CSR_HPMCOUNTER13H = 0xc8d;
+pub const CSR_HPMCOUNTER14H = 0xc8e;
+pub const CSR_HPMCOUNTER15H = 0xc8f;
+pub const CSR_HPMCOUNTER16H = 0xc90;
+pub const CSR_HPMCOUNTER17H = 0xc91;
+pub const CSR_HPMCOUNTER18H = 0xc92;
+pub const CSR_HPMCOUNTER19H = 0xc93;
+pub const CSR_HPMCOUNTER20H = 0xc94;
+pub const CSR_HPMCOUNTER21H = 0xc95;
+pub const CSR_HPMCOUNTER22H = 0xc96;
+pub const CSR_HPMCOUNTER23H = 0xc97;
+pub const CSR_HPMCOUNTER24H = 0xc98;
+pub const CSR_HPMCOUNTER25H = 0xc99;
+pub const CSR_HPMCOUNTER26H = 0xc9a;
+pub const CSR_HPMCOUNTER27H = 0xc9b;
+pub const CSR_HPMCOUNTER28H = 0xc9c;
+pub const CSR_HPMCOUNTER29H = 0xc9d;
+pub const CSR_HPMCOUNTER30H = 0xc9e;
+pub const CSR_HPMCOUNTER31H = 0xc9f;
+pub const CSR_MCYCLEH = 0xb80;
+pub const CSR_MINSTRETH = 0xb82;
+pub const CSR_MHPMCOUNTER3H = 0xb83;
+pub const CSR_MHPMCOUNTER4H = 0xb84;
+pub const CSR_MHPMCOUNTER5H = 0xb85;
+pub const CSR_MHPMCOUNTER6H = 0xb86;
+pub const CSR_MHPMCOUNTER7H = 0xb87;
+pub const CSR_MHPMCOUNTER8H = 0xb88;
+pub const CSR_MHPMCOUNTER9H = 0xb89;
+pub const CSR_MHPMCOUNTER10H = 0xb8a;
+pub const CSR_MHPMCOUNTER11H = 0xb8b;
+pub const CSR_MHPMCOUNTER12H = 0xb8c;
+pub const CSR_MHPMCOUNTER13H = 0xb8d;
+pub const CSR_MHPMCOUNTER14H = 0xb8e;
+pub const CSR_MHPMCOUNTER15H = 0xb8f;
+pub const CSR_MHPMCOUNTER16H = 0xb90;
+pub const CSR_MHPMCOUNTER17H = 0xb91;
+pub const CSR_MHPMCOUNTER18H = 0xb92;
+pub const CSR_MHPMCOUNTER19H = 0xb93;
+pub const CSR_MHPMCOUNTER20H = 0xb94;
+pub const CSR_MHPMCOUNTER21H = 0xb95;
+pub const CSR_MHPMCOUNTER22H = 0xb96;
+pub const CSR_MHPMCOUNTER23H = 0xb97;
+pub const CSR_MHPMCOUNTER24H = 0xb98;
+pub const CSR_MHPMCOUNTER25H = 0xb99;
+pub const CSR_MHPMCOUNTER26H = 0xb9a;
+pub const CSR_MHPMCOUNTER27H = 0xb9b;
+pub const CSR_MHPMCOUNTER28H = 0xb9c;
+pub const CSR_MHPMCOUNTER29H = 0xb9d;
+pub const CSR_MHPMCOUNTER30H = 0xb9e;
+pub const CSR_MHPMCOUNTER31H = 0xb9f;
+pub const CSR_MTVT = 0x307;
+pub const CSR_MNXTI = 0x345;
+pub const CSR_MCOUNTINHIBIT = 0x320;
+pub const CSR_MNVEC = 0x7C3;
+pub const CSR_MTVT2 = 0x7EC;
+pub const CSR_JALMNXTI = 0x7ED;
+pub const CSR_PUSHMCAUSE = 0x7EE;
+pub const CSR_PUSHMEPC = 0x7EF;
+pub const CSR_PUSHMSUBM = 0x7EB;
+pub const CSR_WFE = 0x810;
+pub const CSR_SLEEPVALUE = 0x811;
+pub const CSR_TXEVT = 0x812;
+pub const CSR_MMISC_CTL = 0x7d0;
+pub const CSR_MSUBM = 0x7c4;
+pub const CAUSE_MISALIGNED_FETCH = 0x0;
+pub const CAUSE_FAULT_FETCH = 0x1;
+pub const CAUSE_ILLEGAL_INSTRUCTION = 0x2;
+pub const CAUSE_BREAKPOINT = 0x3;
+pub const CAUSE_MISALIGNED_LOAD = 0x4;
+pub const CAUSE_FAULT_LOAD = 0x5;
+pub const CAUSE_MISALIGNED_STORE = 0x6;
+pub const CAUSE_FAULT_STORE = 0x7;
+pub const CAUSE_USER_ECALL = 0x8;
+pub const CAUSE_SUPERVISOR_ECALL = 0x9;
+pub const CAUSE_HYPERVISOR_ECALL = 0xa;
+pub const CAUSE_MACHINE_ECALL = 0xb;
+
+const VectorTable = microzig.chip.VectorTable;
+
+// will be imported by microzig.zig to allow system startup.
+pub var vector_table: VectorTable = blk: {
```

Comment:
oh do you have to do this because of #64 ? if interrupt implementations are per-chip for riscv then this code shouldn't be here

Reply:
So faced with this problem in addition to removing the tableVector would also need to comment out the existing builds.
Like the stm32f103 it cannot be built at the present time because the project requires interrupts.